src/reader.lisp

```diff
@@ -163,9 +163,8 @@ It ensures the presence of source metadata for STREAM and then calls MAYBE-READ-
   "Compile FORMS as Coalton using the indicated MODE."
   (let* ((*readtable* (named-readtables:ensure-readtable 'coalton:coalton))
          (string (print-form (cons mode forms)))
-         (*source* (unless (source-filename)
```

Comment:
Incorrect conditional: in the context of slime evaluation, source-filename refers to the overall form within which the macro occurs, but we want offsets relative only to the printed form.